### PR TITLE
pouchdb-live-find - type declaration files

### PR DESCRIPTION
**Pull request by assemblethis**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/pouchdb-live-find/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+declare namespace PouchDB {
+    namespace LiveFind {
+        interface RequestDef<Content extends {}> extends Pick<Find.FindRequest<Content>, Exclude<keyof Find.FindRequest<Content>, 'use_index'>> {
+            aggregate?: boolean; // if true outputs an aggregate list on every update event
+        }
+
+        interface PaginateOptions<Content extends {}> extends Pick<Find.FindRequest<Content>, 'sort' | 'skip'| 'limit'> {}
+
+        interface UpdateEvent {
+            action: 'REMOVE' | 'ADD' | 'UPDATE';
```
Comment (rpgeeganage) [inline types/pouchdb-live-find/index.d.ts]: I think that declare an enum for `'REMOVE' | 'ADD' | 'UPDATE'` will be helpful.

File: types/pouchdb-live-find/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+// TypeScript Version: 3.0
+
+/// <reference types="pouchdb-core" />
+declare namespace PouchDB {
+    namespace LiveFind {
+        interface RequestDef<Content extends {}> extends Pick<Find.FindRequest<Content>, Exclude<keyof Find.FindRequest<Content>, 'use_index'>> {
+            aggregate?: boolean; // if true outputs an aggregate list on every update event
+        }
+
+        interface PaginateOptions<Content extends {}> extends Pick<Find.FindRequest<Content>, 'sort' | 'skip'| 'limit'> {}
```
Comment (rpgeeganage) [inline types/pouchdb-live-find/index.d.ts]: I think that declare an enum for 'sort' | 'skip'| 'limit' will be helpful.

File: types/pouchdb-live-find/index.d.ts (hunk 1 above)
Comment (assemblethis) [reply types/pouchdb-live-find/index.d.ts]: From what I've read string literal types are the way to go (https://medium.com/@katbusch/typescript-enums-explained-e5f9a101afc9): 

```
Recommendation
So of the many types of enums which do I recommend using?

Trick question: none.

I recommend string literal types. It’s more TypeScripty. It’s extremely readable. You can see it in the code as is. If you print the value of a variable you won’t get a mysterious 0 or 1; you’ll get the actual string itself every time.

```

File: types/pouchdb-live-find/index.d.ts (hunk 2 above)
Comment (assemblethis) [reply types/pouchdb-live-find/index.d.ts]: These are the keys of the interface that I'm picking. Can you give an example of how you would handle this with an enum?

Comment (rpgeeganage) [reply types/pouchdb-live-find/index.d.ts]: What about something like https://github.com/DefinitelyTyped/DefinitelyTyped/blob/5344bfc80508c53a23dae37b860fb0c905ff7b24/types/pinterest-sdk/index.d.ts#L13

File: types/pouchdb-live-find/index.d.ts (hunk 1 above)
Comment (rpgeeganage) [reply types/pouchdb-live-find/index.d.ts]: some enum like https://github.com/DefinitelyTyped/DefinitelyTyped/blob/5344bfc80508c53a23dae37b860fb0c905ff7b24/types/pinterest-sdk/index.d.ts#L13

**Pull request merged**